src/Sparrow/Json/JsonOperationContext.cs

```diff
@@ -273,7 +269,9 @@ public virtual void Dispose()
 
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public LazyStringValue GetLazyStringForFieldWithCaching(StringSegment key)
-        {
+        {            
+            Debug.Assert(key.Length == key.String.Length, "If the segment do not comprise the whole string this method will fail.");
```

Comment:
This is raised by a bunch of tests (run FastTests).
This also indicate that we need to consider whatever we are actually saving anything by using StringSegment or if we should remove it.